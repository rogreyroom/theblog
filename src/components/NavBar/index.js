import React from 'react'
import { Link } from 'react-router-dom'
import classes from './index.module.scss'

const NavBar = () => (
  <header className={classes.header}>
    <nav className={classes.nav}>
      <Link to="/" className={classes.nav__link}>
        <p className={classes.nav__text}>home</p>
      </Link>
    </nav>
    <Link to="/" className={classes.header__link}>
      <h1 className={classes.header__logo}>TheBlog</h1>
    </Link>
  </header>
)

export default NavBar
