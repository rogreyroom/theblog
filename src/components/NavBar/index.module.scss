.header {
  background-color: var(--c-accent);
  display: grid;
  grid-template-rows: clamp(var(--fs-text-large), 0.7212rem + 0.7692vw,  var(--fs-text-large)) auto;
  grid-template-columns: clamp(17.75rem, -1.9821rem + 92.8571vw, 50.25rem);
  grid-template-areas: 'navigation' 'logo';
  grid-gap: 1rem 0;
  justify-items: end;
  min-height: clamp(12.5rem, 5.6034rem + 34.4828vw, 25rem);
  padding: clamp(var(--m), 0.0905rem + 5.1724vw, var(--xxl));

  @media screen and (min-width: 56.25rem) {
    grid-template-columns: auto 804px auto;
    grid-template-areas: '. navigation .' '. logo .';
  }

    &__link {
      grid-area: logo;
      align-self: end;
      text-decoration: none;
      margin: 0;
    }

    &__logo {
      font-family: var(--ff-logo);
      font-size: clamp(4.5rem, 3.1667rem + 5.9259vw, 6.5rem);
      line-height: clamp(4.5rem, 3.1667rem + 5.9259vw, 6.5rem);
      color: var(--c-black);
      text-shadow: 4px 4px 8px var(--c-black-alpha);
      margin: 0;
    }
}

.nav {
  grid-area: navigation;

  &__link {
    font-size: clamp(var(--fs-text-small), 0.7212rem + 0.7692vw,  var(--fs-text-large));
    line-height: clamp(var(--fs-text-small), 0.7212rem + 0.7692vw,  var(--fs-text-large));
    text-decoration: none;

    &:hover {
      text-decoration: underline 3px var(--c-black);
    }
  }

  &__text {
    color: var(--c-black);
    font-size: clamp(var(--fs-text-normal), 0.7212rem + 0.7692vw,  var(--fs-text-large));
    margin: 0;
  }
}

