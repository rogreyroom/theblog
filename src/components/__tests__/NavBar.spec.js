/**
 * @jest-environment jsdom
 */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { screen, render, cleanup } from '@testing-library/react'
import NavBar from '../NavBar'

describe('NavBar component', () => {
  let element
  beforeEach(() => {
    const { container } = render(
      <Router>
        <NavBar />
      </Router>,
    )
    element = container
  })

  it('renders the component', () => {
    expect(element).toMatchSnapshot()
  })

  it('component contains text "TheBlog" and "home"', () => {
    expect(screen.getByText(/the/i)).toHaveTextContent('TheBlog')
    expect(screen.getByText(/home/i)).toHaveTextContent('home')
  })

  it('renders two link tags', () => {
    const aTag = element.querySelectorAll('a')
    expect(aTag).toHaveLength(2)
  })

  afterAll(cleanup)
})
