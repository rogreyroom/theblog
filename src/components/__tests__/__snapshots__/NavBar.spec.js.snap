// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders the component 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="nav"
    >
      <a
        class="nav__link"
        href="/"
      >
        <p
          class="nav__text"
        >
          home
        </p>
      </a>
    </nav>
    <a
      class="header__link"
      href="/"
    >
      <h1
        class="header__logo"
      >
        TheBlog
      </h1>
    </a>
  </header>
</div>
`;
